### DISCUSSION TITLE
v2.9.3和之后，c++ 的符号剥离机制是不是变化了？gdb 调试 coredump 时，需要保证二进制文件和 sym 文件在同一个目录才能看到符号信息

### DISCUSSION THREAD

**Discussion by bursto7** [xmake-io/xmake#5409] [General]
- 我的版本：xmake 2.9.3
- 问题：从 2.8.x（23 年的版本，具体没有记录）升级到 2.9.3 后，在 releasedbg 模式下抽离符号 sym 文件，应用于gdb 调试 core 文件时，看不到符号信息，显示问号。**_而原本我只需要 `gdb *.sym core-filename` 即可打开看到对应的符号信息_**。现在，我需要确保二进制和 sym 在同一个目录，然后执行 gdb 二进制才能打开。另外，我无法先执行 `gdb 二进制文件`，然后在 gdb 交互窗口中通过`symbol-file symfilepath`完成符号加载，通过这种方式依旧显示问号。
- 求助：既然无法直接通过 `gdb symfile` 进行 debug，那么是否能通过 symbol-file 进行加载？因为我们是一个大型项目，实际部署和调试时，需要分别存放二进制文件和 sym。

补充信息，我们的 release 做了配置：
```
if is_mode("releasedbg") then
    add_cxflags("-fno-omit-frame-pointer")
    set_strip("debug") 
end
```
我已经修改为，但没有变化：
```
if is_mode("releasedbg") then
    add_cxflags("-fno-omit-frame-pointer")
    set_strip("all") -- strip -g
    set_symbols("debug")
end
```

**Comment by bursto7** [comment]
找到变化的 issue 了：https://github.com/xmake-io/xmake/issues/4684

我的编译log：
```
[ 99%]: generating.releasedbg hello.sym
/opt/rh/gcc-toolset-12/root/usr/bin/objcopy --only-keep-debug build/linux/x86_64/releasedbg/hello build/linux/x86_64/releasedbg/hello.sym
/opt/rh/gcc-toolset-12/root/usr/bin/strip -s build/linux/x86_64/releasedbg/hello
/opt/rh/gcc-toolset-12/root/usr/bin/objcopy --add-gnu-debuglink=build/linux/x86_64/releasedbg/hello.sym build/linux/x86_64/releasedbg/hello
```

**Comment by waruqi** [comment]
> 既然无法直接通过 gdb symfile 进行 debug，那么是否能通过 symbol-file 进行加载？因为我们是一个大型项目，实际部署和调试时，需要分别存放二进制文件和 sym。

其实之前那个 symfile 不算是纯符号文件。。是完整带 `-g` 符号的可执行文件，所以你能直接调试。你自己加 `-g` 生成的可执行文件就是这个，直接 gdb 加载执行就行了。
